app/src/main/java/com/ncnf/database/DatabaseService.java

```diff
@@ -237,5 +247,41 @@ private void initRegistry(){
     public CompletableFuture<Boolean> removeArrayField(String documentPath, String arrayField, Object value){
         return this.updateField(documentPath, arrayField, FieldValue.arrayRemove(value));
     }
+
+    public CompletableFuture<List<SocialObject>> eventGeoQuery(LatLng location, double radius){
+        radius = (radius < 1000) ? radius * 1000 : radius; //Check if radius is still in km, convert to m
+
+        List<GeoQueryBounds> bounds = GeoFireUtils.getGeoHashQueryBounds(new GeoLocation(location.latitude, location.longitude), radius);
+        final List<Task<QuerySnapshot>> tasks = new ArrayList<>();
+        for (GeoQueryBounds b : bounds){
+            Query q = db.collection(EVENTS_COLLECTION_KEY)
+                    .orderBy(GEOHASH_KEY)
+                    .startAt(b.startHash)
+                    .endAt(b.endHash);
+            tasks.add(q.get());
+        }
+
+        CompletableFuture<List<SocialObject>> futureResponse = new CompletableFuture<>();
+
+        Tasks.whenAllComplete(tasks)
+                .addOnCompleteListener(task -> {
+                    if (task.isSuccessful()) {
+                        List<DocumentSnapshot> matchingDocs = new ArrayList<>();
+                        List<SocialObject> result = new ArrayList<>();
+                        for (Task<QuerySnapshot> t : tasks) {
+                            QuerySnapshot snap = t.getResult();
+                            matchingDocs.addAll(snap.getDocuments());
+                        }
+                        for (DocumentSnapshot doc : matchingDocs){
+                            result.add((Event) registry.get(Event.class).toObject(doc.getId(), doc.getData()));
+                        }
+                        futureResponse.complete(result);
+                    } else {
+                        futureResponse.completeExceptionally(task.getException());
+                    }
+                });
+
+        return futureResponse;
+    }
 }
```

Comment:
If the method is specific to a social object, it should not be in DatabaseService which is generic. 
Put the method in another class (SocialObject or other). You can also try to generify a part of your query such that it can be used with other collections that have a Geo field.